src/lib/MetalCam.mm

```diff
@@ -148,7 +148,7 @@ - (void) update {
         [self _updateOpenGLTexture];
     }
 
-#if defined(__IPHONE_13_0) && defined(ARBodyTrackingBool_h)
+#if defined(__IPHONE_13_0)
```

Comment:
So the one issue here is that, what if the user's device does support body tracking but they haven't set up the proper configuration type? 

Should probably be written like this (note - not tested)

```javascript
#if defined(__IPHONE_13_0)
if(ARBodyTrackingConfiguration.isSupported && [_session.configuration isKindOfClass:[ARBodyTrackingConfiguration class]]){
        // update depth textures
        [self _updateMatteTextures: commandBuffer];
    }
#endif
```

I think I spot a few other instances that are similar. If you happen to get to it, fantastic, otherwise don't worry about it, I'll make the necessary changes before I merge this.

Reply:
you are looking at the wrong file, i did a push `ac02256` , removing it.
Although did leave one in `ARProcessor`

Reply:
Actually `line 200` in `ARSessionSetup.h` : 
`                if(state.useARFrameSemanticPersonSegmentationWithDepth && [ARBodyTrackingConfiguration isSupported] ){
`

created `useARFrameSemanticPersonSegmentationWithDepth` for that.

Reply:
also : 
           ` if(this->session.configuration.frameSemantics == ARFrameSemanticPersonSegmentationWithDepth){` is a valid condition, but the `config` is a `ARWorldTrackingConfiguration` so body tracking cast check isn't working.